backends/jgroups/src/main/java/org/hibernate/search/backend/jgroups/impl/DispatchMessageSender.java

```diff
@@ -71,68 +72,58 @@ public View getView() {
 	}
 
 	@Override
-	public void send(final Message message, final boolean synchronous, final long timeout) throws Exception {
+	public void send(final Buffer data, final boolean synchronous, final long timeout) throws Exception {
 		final RequestOptions options = synchronous ? RequestOptions.SYNC() : RequestOptions.ASYNC();
-		options.setExclusionList( dispatcher.getChannel().getAddress() );
+		options.exclusionList( dispatcher.getChannel().getAddress() );
+		options.setTransientFlags( TransientFlag.DONT_LOOPBACK );
```

Comment:
Are **both** of those options necessary? In my mind excluding the local address and preventing loopback are the same, but I may be missing something...

Reply:
In most cases you'll only need one but they are subtly different depending on the protocols being used.

the exclusion list is applied on the packet headers, like destination delivery with UDP anycast.
The "loopback" option affects the loop delivery, so if a message is broadcasted - it's still received by the local network interface but we have JGroups "not-deliver" it back up again.